Source/WTF/wtf/NativePromise.h

```diff
@@ -713,25 +721,35 @@ class NativePromise final : public NativePromiseBase, public ConvertibleToNative
             ASSERT(!promise.isNothing());
 
             if (UNLIKELY(!m_targetQueue || (promise.m_dispatchMode == PromiseDispatchMode::RunSynchronouslyOnTarget && m_targetQueue->isCurrent()))) {
-                PROMISE_LOG(*promise.m_result ? "Resolving" : "Rejecting", " synchronous then() call made from ", m_logSiteIdentifier, "[", promise, " callback:", (const void*)this, "]");
                 if (m_disconnected) {
-                    PROMISE_LOG("ThenCallback disconnected aborting [callback:", (const void*)this, " callSite:", m_logSiteIdentifier, "]");
+                    PROMISE_LOG("ThenCallback disconnected from ", promise, " aborting [callback:", (const void*)this, " callSite:", m_logSiteIdentifier, "]");
                     return;
                 }
                 {
                     // Holding the lock is unnecessary while running the resolve/reject callback and we don't want to hold the lock for too long.
                     DropLockForScope unlocker(lock);
-                    processResult(promise.result());
+                    if (promise.hasRunnable()) {
+                        ASSERT(IsExclusive);
+                        auto resultRunnable = WTFMove(promise.resultRunnable());
```

Comment:
This WTFMove is a bit odd.
Should we encapsulate this in the `result()` call instead that would call resultRunnable() and store the returned value in m_result?
Or have a takeResultRunnable() maybe.

Reply:
to do so would require a more intensive change to the code rather than re-use what's already there as much as possible

Reply:
As discussed off-line; I have made the move semantics a bit clearer , with the use of a takeResult/result and takeResultRunnable